### PR TITLE
Feature/ask before deleting not empty list

### PR DESCRIPTION
**Pull request by RedX2501**
Will ask if it is ok to delete the list if it still has items.

The function as factored out so I can later use it in the shopping lists overview when deleting shopping lists from there.
Please let me know if that is fine or you'd like to have a component.

**This PR is still a draft due to it's dependency on other PRs.** I'd like to get your feedback early so I know if the style is good.

### CODE REVIEW

**Comment by RedX2501** [comment]
[] do not merge as it depends on other PRs.

**Comment by TomBursch** [comment]
Looks good already, thanks for this. If you have the time, can you split the documentation into its own PR? Thanks!

Edit: Ah, sorry, went through this one first and just saw that you did that already 👍🏼

File: kitchenowl/lib/l10n/app_en.arb [hunk 1]
```diff
@@ -580,6 +587,7 @@
   "shoppingList": "Shopping list",
   "shoppingListDelete": "Delete shopping list",
   "shoppingListDeleteConfirmation": "Are you sure you want to delete {shoppingList}?",
+  "shoppingListContainsEntries": "{entriesCount, plural, =0{The shopping list contains no entrie(s).} =1{The shopping list contains 1 entry.} other{The shopping list contains {entriesCount} entries.}}",
```
Comment (TomBursch) [inline kitchenowl/lib/l10n/app_en.arb]: Can we order these alphabetically? Just move this up two lines. Same for the `@shoppingListContainsEntries`.

**Review by TomBursch — CHANGES_REQUESTED** [review]
Only one minor thing, otherwise it LGTM

File: kitchenowl/lib/l10n/app_en.arb (hunk 1 above)
Comment (RedX2501) [reply kitchenowl/lib/l10n/app_en.arb]: Done

**Review by TomBursch — APPROVED** [review]

**Pull request merged**